mzmine-community/src/main/java/io/github/mzmine/util/scans/similarity/impl/ms2deepscore/SettingsMS2Deepscore.java

```
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2004-2024 The mzmine Development Team
+ *
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following
+ * conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ * OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package io.github.mzmine.util.scans.similarity.impl.ms2deepscore;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public record SettingsMS2Deepscore(@JsonProperty("embedding_dim") Integer embeddingDimension,
+                                   @JsonProperty("ionisation_mode") String ionisationMode,
+                                   @JsonProperty("min_mz") int minimumMZ,
+                                   @JsonProperty("max_mz") int maximumMZ,
+                                   @JsonProperty("mz_bin_width") Double binWidth,
+                                   @JsonProperty("additional_metadata") Object[][] additionalMetadata,
+                                   @JsonProperty("intensity_scaling") float intensityScaling
```

just a more general comment. You can slab a naming convention on this and it will automatically translate from camelCase to snake_case.

Put this before the class - but then your names need to directly match, e.g., mzBinWidth --> mz_bin_width:
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)

Also you can default ignore or fail on unknown properties
@JsonIgnoreProperties(ignoreUnknown = true)